dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java

```diff
@@ -626,6 +631,77 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
 
     }
 
+    @DataProvider
+    public static List<Object> dataProviderEEContentTypes() throws Exception {
+        // data provider needs stuff to get initialized because of API access
+        IntegrationTestInitService.getInstance().init();
+
+        // filter only Enterprise content types
+        List<ContentType> eeTypes = APILocator
+                .getContentTypeAPI(APILocator.systemUser()).findAll().stream()
+                .filter((type)->type instanceof EnterpriseType).collect(Collectors.toList());
+
+        // returns a List of Tuple (typeName, baseType)
+        return eeTypes.stream().map((type)->
+                new Tuple2<>("my"+type.variable(), type.baseType())
+        ).collect(Collectors.toList());
+    }
+
+    @Test
+    @UseDataProvider("dataProviderEEContentTypes")
+    public void testGetSchema_GivenNoEELicense_EnterpriseTypesShouldNotBeAvailableInSchema(
+            final Tuple2<String, BaseContentType> testCase) throws Exception{
+        ContentType customType = null;
+
+        try {
+            // create custom persona type. 1=typeName, 2=BaseType
+            customType = createType(testCase._1,
+                    testCase._2);
+
+            runNoLicense(() -> {
+                final GraphQLSchema schema = APILocator.getGraphqlAPI().getSchema();
+                assertNull(schema.getType(testCase._1));
+            });
+        } finally {
+            if(customType!=null) {
+                APILocator.getContentTypeAPI(APILocator.systemUser()).delete(customType);
+            }
+        }
+    }
+
+    @DataProvider
+    public static List<Object> dataProviderEEBaseTypes() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4740432)